tools/system_libs.py

```diff
@@ -1867,6 +1867,22 @@ def can_use(self):
     return settings.WASMFS
 
 
+class libwasmfs_noderawfs(Library):
+  name = 'libwasmfs_noderawfs'
+
+  cflags = ['-fno-exceptions', '-std=c++17']
+
+  includes = ['system/lib/wasmfs']
+
+  def get_files(self):
+    return files_in_path(
+        path='system/lib/wasmfs/backends',
+        filenames=['noderawfs_root.cpp'])
```

Comment:
I would prefer not to add another native system library just for this purpose.

Can we find another way (other than weak linking) to express the mount point?

Reply:
Sorry, I should have asked you for review probably before landing.

I'm not sure there is a better mechanism. We don't want to bloat code size by having some kind of global dispatch "if X then root backend = .. else if Y then root backend = ..". The only slim (and extensible) way to do it that I know is to have a weak symbol that is overridden like this. I can't think of another way at least.

Reply:
An alternative might be to use `EM_JS` to check whether the `NODERAWFS` setting is set, for example:
https://github.com/kleisauke/wasm-vips/blob/cff9f510dcfba6cc5bb8cb219e1ce140959ff4f8/src/vips-emscripten.cpp#L45-L53

However, in terms of code size, this mechanism is probably better.

Reply:
Or, perhaps the `get_default_variation()` mechnasim would be better?:

<details>
  <summary>Details</summary>

```diff
--- a/tools/system_libs.py
+++ b/tools/system_libs.py
@@ -1821,6 +1821,7 @@ class libwasmfs(DebugLibrary, AsanInstrumentedLibrary, MTLibrary):
 
   def __init__(self, **kwargs):
     self.ignore_case = kwargs.pop('ignore_case')
+    self.node_rawfs = kwargs.pop('node_rawfs')
     super().__init__(**kwargs)
 
   def get_cflags(self):
@@ -1833,6 +1834,8 @@ class libwasmfs(DebugLibrary, AsanInstrumentedLibrary, MTLibrary):
     name = super().get_base_name()
     if self.ignore_case:
       name += '-icase'
+    if self.node_rawfs:
+      name += '-noderawfs'
     return name
 
   @classmethod
@@ -1841,7 +1844,7 @@ class libwasmfs(DebugLibrary, AsanInstrumentedLibrary, MTLibrary):
 
   @classmethod
   def get_default_variation(cls, **kwargs):
-    return super().get_default_variation(ignore_case=settings.CASE_INSENSITIVE_FS, **kwargs)
+    return super().get_default_variation(ignore_case=settings.CASE_INSENSITIVE_FS, node_rawfs=settings.NODERAWFS, **kwargs)
 
   def get_files(self):
     backends = files_in_path(
@@ -1852,6 +1855,10 @@ class libwasmfs(DebugLibrary, AsanInstrumentedLibrary, MTLibrary):
                    'memory_backend.cpp',
                    'node_backend.cpp',
                    'opfs_backend.cpp'])
+    if self.node_rawfs:
+        backends += files_in_path(
+            path='system/lib/wasmfs/backends',
+            filenames=['noderawfs_root.cpp'])
     return backends + files_in_path(
         path='system/lib/wasmfs',
         filenames=['file.cpp',
@@ -1867,22 +1874,6 @@ class libwasmfs(DebugLibrary, AsanInstrumentedLibrary, MTLibrary):
     return settings.WASMFS
 
 
-class libwasmfs_noderawfs(Library):
-  name = 'libwasmfs_noderawfs'
-
-  cflags = ['-fno-exceptions', '-std=c++17']
-
-  includes = ['system/lib/wasmfs']
-
-  def get_files(self):
-    return files_in_path(
-        path='system/lib/wasmfs/backends',
-        filenames=['noderawfs_root.cpp'])
-
-  def can_use(self):
-    return settings.WASMFS and settings.NODERAWFS
-
-
 class libhtml5(Library):
   name = 'libhtml5'
 
```
(untested)

</details>

Reply:
> An alternative might be to use EM_JS [..] However, in terms of code size, this mechanism is probably better.

Yeah, exactly - EM_JS can work, and linking is simpler, but it would be more code that can't be removed by the linker, since it won't be able to tell what the EM_JS call will return. Using weak linking, that information is utilized at link time basically.

I don't have a strong opinion on using `get_default_variation()`, but the current code seems easier to understand to me at least...